src/mono/netcore/System.Private.CoreLib/src/System/Delegate.Mono.cs

```diff
@@ -214,7 +214,7 @@ public static Delegate CreateDelegate(Type type, object? firstArgument, MethodIn
             return CreateDelegate_internal(type, null, info, throwOnBindFailure);
         }
 
-        private static MethodInfo? GetCandidateMethod(RuntimeType type, [DynamicallyAccessedMembers(AllMethods)] Type target, string method, BindingFlags bflags, bool ignoreCase)
+        private static MethodInfo? GetCandidateMethod(RuntimeType type, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type target, string method, BindingFlags bflags, bool ignoreCase)
```

Comment:
@vitek-karas @sbomer Looks like this is where the limitation of the illink analysis actually bit us - the loop is over `.BaseType`, but I assume because of how the backwards branch ends up happening, linker missed the fact that GetMethod could also be called on the base type. In NativeAOT this is a `while` loop that is probably generated differently.

We should probably track (either in our heads or in an issue) how often we hit something like this.

Reply:
Ah, just found https://github.com/mono/linker/issues/1755 @eerhardt filed on this. I'm pretty sure the underlying issue is the backwards branch.